pkg/apis/v1beta1/cluster.go

```diff
@@ -74,24 +82,51 @@ func (s *SchedulerSpec) IsZero() bool {
 	return len(s.ExtraArgs) == 0
 }
 
+func (c *SchedulerSpec) Validate() []error {
+	return nil
+}
+
+var _ Validator = (*InstallSpec)(nil)
+
 // InstallSpec defines the required fields for the `k0s install` command
 type InstallSpec struct {
 	SystemUsers *SystemUser `yaml:"users,omitempty"`
 }
 
+// Validate stub for Validator interface
+func (i *InstallSpec) Validate() []error {
+	return nil
+}
+
+// Validator interface to ensure that all config components implement Validate function
+type Validator interface {
```

Comment:
hmm, I'd maybe call this something like `Validateable` as this ensures we can validate such an interface